lib/lottery/megamillion.js

```diff
@@ -0,0 +1,125 @@
+const axios = require('axios');
+
+const options = {
+  method: 'GET',
+  url: 'https://www.megamillions.com/cmspages/utilservice.asmx/GetLatestDrawData',
+};
+
+async function makeMegamillionCall() {
+  const response = await axios.request(options)
+  return response.data;
+}
+
+function parseRawData(rawData) {
+  let start = rawData.indexOf('{');
+  let end;
+  if (start < 0) {
+    return '';
+  }
+  for (let i = rawData.length -1; i > 0; i--) {
```

Comment:
I found there is a function we can use here: String.prototype.lastIndexOf()

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/lastIndexOf

Reply:
added